rules-tests/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector/Fixture/if_check_with_empty_multiple.php.inc

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\If_\RemoveUnusedNonEmptyArrayBeforeForeachRector\Fixture;
+
+final class IfCheckWithEmptyMultiple
+{
+    public function run(array $items, array $items2)
+    {
+        if (empty($items)) {
+            return;
+        }
+
+        foreach ($items as $item) {
+            echo $item;
+        }
+
+        if (empty($items2)) {
+            return;
+        }
+
+        foreach ($items2 as $item2) {
+            echo $item2;
+        }
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\If_\RemoveUnusedNonEmptyArrayBeforeForeachRector\Fixture;
+
+final class IfCheckWithEmptyMultiple
+{
+    public function run(array $items, array $items2)
+    {
+        foreach ($items as $item) {
+            echo $item;
+        }
+
+        foreach ($items2 as $item2) {
+            echo $item2;
+        }
```

Comment:
This fixture should be untouched actually.

If `$items` is empty, it will never continue iterate the `$items2`, even if `$items2` have some values.

Reply:
I see, but `isMatchingEmptyAndForeachedExpr()` need to be kept when it doesn't return, eg:

```php
        if (empty($items)) {
             throw new Exception('test');
        }

        foreach ($items as $item) {
            echo $item;
        }
```

or just echo it:

```php
        if (empty($items)) {
            echo "Warning: items is empty";
        }

        foreach ($items as $item) {
            echo $item;
        }
```

I will revert my changes and add more fixtures.

Reply:
This rule only handle empty followed by foreach, nothing else like throw die etc.

Reply:
I reverted my changes, I will add fixture for throw Exception or just echo inside empty check.

Reply:
The `$this->uselessIfCondBeforeForeachDetector->isMatchingEmptyAndForeachedExpr()` by pass that, as only check if pass `empty()`, then foreach next, I will add fixture for that.

Reply:
I added failing fixture for throw inside empty check: https://github.com/rectorphp/rector-src/pull/3611/commits/2f298aab531087f900b2d3f8ffa7e9f2a89031e5 , it show the diff:

```diff
     public function run(array $items)
     {
-        if (empty($items)) {
-            throw new Exception('items must not be empty');
-        }
-
         foreach ($items as $item) {
             echo $item;
         }
```

which should be skipped.